### PR TITLE
Load and validate SGX quote

### PR DESCRIPTION
**Pull request by definitelynobody**

### CODE REVIEW

**Comment by definitelynobody** [comment]
Both haralds initial quote and the icelake quote validate, however the is_trusted function does not trust the icelake leaf certificate.

File: src/ext/sgx/mod.rs [hunk 1]
```diff
@@ -1,47 +1,44 @@
...
 
 use super::ExtVerifier;
 
 #[derive(Clone, Debug, PartialEq, Eq, Sequence)]
 pub struct Evidence<'a> {
-    pub pck: Certificate<'a>,
-
     #[asn1(type = "OCTET STRING")]
     pub quote: &'a [u8],
 }
```
Comment (npmccallum) [inline src/ext/sgx/mod.rs:21]: If we truly only have the quote, then we can throw away the `Evidence` structure and just get the quote directly from the extension encoded bytes.

File: src/ext/sgx/quote/header.rs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    /// contain a QE identifier used to link a PCK Cert to an Enc(PPID). This
+    /// identifier is consistent for every quote generated with this QE on this
+    /// platform.
+    pub user_data: [u8; 20],
+}
+
+impl Default for QuoteHeader {
+    fn default() -> Self {
+        Self {
+            version: QUOTE_VERSION,
```
Comment (npmccallum) [inline src/ext/sgx/quote/header.rs:75]: I'm guessing we probably want an untagged union on quotes. First we look at just the version field. Then we decide what the rest of the structure should be.

File: src/ext/sgx/quote/header.rs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    pub qe_svn: u16,
+    /// Security version of the Provisioning Certification Enclave.
+    pub pce_svn: u16,
+    /// ID of the QE vendor.
+    pub qe_vendor_id: [u8; 16],
+    /// Custom user-defined data. For the Intel DCAP library, the first 16 bytes
+    /// contain a QE identifier used to link a PCK Cert to an Enc(PPID). This
+    /// identifier is consistent for every quote generated with this QE on this
+    /// platform.
+    pub user_data: [u8; 20],
```
[the remark this answers is not in the document]
Comment (definitelynobody) [reply src/ext/sgx/quote/header.rs]: This is definitely a big improvement. In some structs which are dynamically sized I didn't know how to do this but I did remove a lot of copying.

File: src/ext/sgx/quote/header.rs (hunk 2 above)
Comment (definitelynobody) [reply src/ext/sgx/quote/header.rs]: I switched to using an enum because the code was a lot easier to use.

**Comment by definitelynobody** [comment]
blocked by https://github.com/enarx/sgx/pull/75

**Comment by definitelynobody** [comment]
https://github.com/enarx/sgx/pull/78

**Comment by npmccallum** [comment]
@definitelynobody Please add `sgx` as a git dependency so CI can work. The steward doesn't get published in crates.io yet. So this is acceptable (for now).

**Comment by npmccallum** [comment]
@definitelynobody Also, can you submit the first commit as a separate PR so I can merge it immediately? I think we probably need to reorganize the commits so that they are logical and self-contained rather than stream of consciousness style.

**Comment by npmccallum** [comment]
@bstrie Can you please take a review pass at this?

**Comment by definitelynobody** [comment]
pushed some fixes for the sgx changes

**Comment by npmccallum** [comment]
@definitelynobody Merged with other fixes and full attestation.

**Pull request closed without merging**